### PR TITLE
Disallow models with `--arrays-weak-equiv`

### LINKED ISSUE #2007: Add error message when using --arrays-weak-equiv with models
Currently, we get an assertion failure but we should report an error instead.

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Fixes #2007. We currently do not support generating models when
`--arrays-weak-equiv` is enabled. This commit adds a corresponding
user-facing error message.

### CODE REVIEW

File: test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+; COMMAND-LINE: --check-models --arrays-weak-equiv
```
Comment (ajreynol) [inline test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2:1]: Is this configuration necessary?  I'd prefer keeping `check-models` off command line in regressions.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2 (hunk 1 above)
Comment (4tXJ7f) [reply test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2:1]: As opposed to just running with `--produce-models`? I'd be fine with that.

Comment (ajreynol) [reply test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2:1]: Yes, you already have `--produce-models --arrays-weak-equiv` in the line below.

Comment (4tXJ7f) [reply test/regress/regress0/arrays/issue8103-1-weak-equiv-models.smt2]: Yeah, the idea was to make sure that we catch it in both cases. It's not super important, though. I've removed it.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**